pennylane/io/qasm_interpreter.py

```diff
@@ -130,24 +140,53 @@ def update_var(
         value: any,
         name: str,
         node: QASMNode,
-    ):
+    ):  # pylint: disable=too-many-branches
```

Comment:
I know this was done in a previous PR, but now I'm actually thinking having `update_var` take `node.op` and `node.span.start_line` instead. This might help it stay a bit more cleanly separated so that `QasmInterpreter` handles the nodes and `Context` doesn't have to know anything about nodes.  

What would you think?